lib/parser/caffe/caffe_parser.cc

```diff
@@ -129,27 +129,33 @@ Status CAFFEParser::ConvertToHaloIR(
   }
   auto layer_size_weight = weight_v1 ? net_param_weight.layers_size()
                                      : net_param_weight.layer_size();
-  auto is_input_layer = [&](int i) {
-    if (weight_v1) {
-      return net_param_weight.layers(i).type() == caffe::V1LayerParameter::DATA;
+
+  std::unordered_map<std::string, int> weight_name_to_id;
+  for (int id = 0; id < layer_size_weight; ++id) {
+    auto name = weight_v1 ? net_param_weight.layers(id).name()
+                          : net_param_weight.layer(id).name();
+    if (weight_name_to_id.count(name)) {
+      std::cerr << "caffemodel has duplicated name: " << name << "\n";
```

Comment:
Need to abort/fail the parsing if a duplicated name is found?

Reply:
> Need to abort/fail the parsing if a duplicated name is found?

Right. Will fix.